From 1ba6e44219f57957cf2ac3573f2204fcd0809a24 Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Wed, 17 Jan 2024 16:43:26 +0100
Subject: [PATCH 0795/1141] memory: stm32: Add JEDEC flashes support

Add jedec_flash boolean which allows to select the calibration
algorithm between CFI and JEDEC flashes.

Change-Id: I679f3e4513dcbac6ee19dc4bc950f12a5d71db22
Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/353483
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
---
 drivers/memory/stm32-omi.c | 7 +++++++
 include/memory/stm32-omi.h | 1 +
 2 files changed, 8 insertions(+)

diff --git a/drivers/memory/stm32-omi.c b/drivers/memory/stm32-omi.c
index 72eda1807a11..9fda9ab80012 100644
--- a/drivers/memory/stm32-omi.c
+++ b/drivers/memory/stm32-omi.c
@@ -435,6 +435,7 @@ static int stm32_omi_probe(struct platform_device *pdev)
 	u8 spi_flash_count = 0;
 	u8 child_count = 0;
 	int ret;
+	bool jedec_flash = false;
 
 	/*
 	 * Flash subnodes sanity check:
@@ -446,6 +447,11 @@ static int stm32_omi_probe(struct platform_device *pdev)
 		if (of_device_is_compatible(child, "cfi-flash"))
 			hyperflash_count++;
 
+		if (of_device_is_compatible(child, "jedec-flash")) {
+			hyperflash_count++;
+			jedec_flash = true;
+		}
+
 		if (of_device_is_compatible(child, "jedec,spi-nor") ||
 		    of_device_is_compatible(child, "spi-nand") ||
 		    of_device_is_compatible(child, "st,octal-bus"))
@@ -480,6 +486,7 @@ static int stm32_omi_probe(struct platform_device *pdev)
 	if (!omi)
 		return -ENOMEM;
 
+	omi->jedec_flash = jedec_flash;
 	omi->regs_base = devm_platform_get_and_ioremap_resource(pdev, 0, &res);
 	if (IS_ERR(omi->regs_base))
 		return PTR_ERR(omi->regs_base);
diff --git a/include/memory/stm32-omi.h b/include/memory/stm32-omi.h
index 8ebd788e07ad..5799e3640398 100644
--- a/include/memory/stm32-omi.h
+++ b/include/memory/stm32-omi.h
@@ -175,6 +175,7 @@ struct stm32_omi {
 	u32 dlyb_base;
 	int irq;
 	bool calibration;
+	bool jedec_flash;
 
 	int (*check_transfer)(struct stm32_omi *omi);
 };
-- 
2.39.2

